### DISCUSSION TITLE
v0.8

### DISCUSSION THREAD

**Discussion by U-C-S** [U-C-S/Hurl#74] [Announcements]
Note: Release Notes and Docs are yet to be updated.

**Full Changelog**: https://github.com/U-C-S/Hurl/compare/v0.7.1...v0.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/U-C-S/Hurl/releases/tag/v0.8.0'>v0.8</a>.</em>

**Comment by AaronFeledy** [comment]
On Windows 11, I had to update my .NET Desktop Runtime to  7.0.9 or later. Running `winget install Microsoft.DotNet.DesktopRuntime.7` took care of it. Perhaps the installer could do this automatically. I'll open a ticket.

**Reply by AaronFeledy** [reply]
#75

**Reply by U-C-S** [reply]
Actually, Depending on winget for automatically installing Runtime looks much better idea than the idea I had. I will test this out to include a script within installer to run it at the post.... Thanks

**Comment by eidylon** [comment]
Just want to say, with the 0.8 release, and the implementation of automatic rules, HURL's feature-set is `PERFECT` now! Works beautifully!

**Comment by flawrist** [comment]
By using NET 7 you effectively cut off Windows 7 users, but why? Come on, think of a solution for a broader audience, not just 'my app is so cool as it uses the latest shiny libraries'. Obviously, there is nothing this app needs from Windows 10 what doesn't exist in Windows 7. For example, Powershell 7.2.17 (built on 2023-11-16) uses .NET 6.0.417 and it works fine here.

**Reply by U-C-S** [reply]
@sergeevabc sorry to disappoint you there. Best I can do for you is release a untested .NET 6 version alongside every minor version (can't promise that it's gonna work fine with win7 or 8). PowerToys is my reasoning and base to upgrade to newer versions. (no specific reason)

**Reply by skyhirider** [reply]
Windows 7 does not seem to have public security update support, and not sure if (or how long) the paid extended security support will last (https://learn.microsoft.com/en-us/troubleshoot/windows-client/windows-7-eos-faq/windows-7-extended-security-updates-faq). Just did a quick web search, so if I am wrong please correct me.

I would say adding support to win7 seems like a waste of time. Unless you have stats that show you have a large user base on 7.

But honestly, running an OS without security updates is not the way to go.

**Reply by flawrist** [reply]
@skyhirider, sorry to hear that you have been so brainwashed with the 'security updates' mantra that you no longer notice how you sound like a drug addict who depends on the next dose and convinces others that this is what real life is.

You see, when a thing is done well, it needs less and less attention from the manufacturer. A good refrigerator continues to cool meat and ice cream even if the warranty has expired or the factory where it was made has closed, Some manufacturers don't want you to forget about them, so you must be constantly kept in fear and constantly comforted with updates. This manipulation is not new, it is used annually in the fashion world, it is called [artificial obsolescence](https://en.wikipedia.org/wiki/Planned_obsolescence).

If you look at how programs were created before, and they were created to last, you will find that you can safely run DOS word processor like the [author of Game of Thrones](https://www.bbc.com/news/technology-27407502) does or Doom the game released a couple of decades ago. Also try visiting places like rural schools, parishes, libraries, lighthouses and even observatories — do they care about hackers lurking around the corner or they appreciate, for example, a more user-friendly interface that the next OS cannot offer? After all, the world is much more diverse than you think. Then maybe you might grasp why there are communities where they cherish not only Windows 7 or XP, but even [Windows 98](https://msfn.org/board/topic/105936-last-versions-of-software-for-windows-98se/). Or at least you might grasp why security tools such as KeePass and GPG still work under Windows 7.  Because some of us still create solutions to ease people's burdens, rather than trying to get them hooked on the needle of updates.

**Reply by skyhirider** [reply]
@sergeevabc I agree that when software is written well, it can last decades. But when its consumer software (where consumers are not security experts) having no updates eventually leads to a very insecure experience. Yes, the UI may be great, but if it gets frequently hacked and you need to reformat its not a great experience overall.

I have nothing against people using older software, its their choice. I was simply saying that in general, there is a reason companies deprecate devices ( such as phones) when they stop receiving updates. Anything connected to the network is vulnerable, as anything man made will have at least minor bugs, not to mention it may be using old components or insecure encryptions protocols that become obsolete as time passes.

I would definitely not trust a WAN router that is running 10 year old software. It is why SMS 2FA is considered insecure, the towers themselves are using old software.

Having the latest greatest however is not a panacea, I fully agree on that. There is a sweet spot with getting the latest security updates while delaying feature updates until they are tested by a wider audience (delaying major updates).

But I would not wish to argue here as this discussion is about the very interesting tool UCS has created, lets simply agree to disagree.

Ultimately, it is up to UCS to decide, I just wanted to provide a different opinion so that the discussion was balanced.

**Reply by flawrist** [reply]
@skyhirider, look, life is vulnerable. And if we follow your logic, then we should live in a bomb shelter that could withstand a meteorite strike and go outside only with weapons, in a gas mask, with self-purified water and dried herbs instead of drugs with unknown chemical formulas. But people don't do that because not everyone is [infected with fear](https://iv.nboeck.de/watch?v=w3H_xsdr-Vs) like you are.

In theory, all sorts of things are possible. For example, physicists are arguing about how many universes there are, and have already piled up a bunch of theories. But in practice, people are guided by a [threat model](https://ssd.eff.org/module/your-security-plan) that relies on the principle of necessity and sufficiency. This can be explained using a calculator.

What updates does a calculator need? None. However, there is a theoretical possibility that one day some black hat will discover a math sequence that causes a floating-point operation to fail, thereby allowing access to, for example, disk formatting. But that's one vulnerability per decade, which, once known, will be fixed, if not by the manufacturer, then by enthusiasts. Not to mention that someone would need to convince me to type that math sequence in the first place. Does this mean that even a calculator can't be trusted and should be run in a sandbox? If you work in a high-risk area like Edward Snowden, the answer is yes, but for everyone else the answer is no. 

As a developer, you should focus on making sure your app helps as many people as possible, because computers were invented for this very purpose — to ease the burden. And you can cut people off from the app for one reason only: “Dear users, it is impossible to compile my app for your OS because it lacks features that my app requires by design to work properly”. To cut people off just because some American businessmen, eager to profit from selling updates, declared some OS obsolete, and not because it lacks some features your app needs, is another kind of discrimination.

**Reply by U-C-S** [reply]
Sorry, have to temporarily lock this conversation. While I get the point, This is going somewhere else. Hope you understand.
